### PR TITLE
Version 0.25.2

### PR DESCRIPTION
**Pull request by karpetrosyan**
# Changelog

### Added

* Add missing type hints to few `__init__()` methods. (#2938)

### CODE REVIEW

**Comment by tomchristie** [comment]
Great. We *could* also mention that we're enforcing `httpcore>=1.0` now, but I don't think it matters too much either way.

Happy to help get this released if needed.

**Review by tomchristie — APPROVED** [review]

**Comment by karpetrosyan** [comment]
We can also consider adding #2879 to this release if you wish.

I think it would be better to start migration to the new proxy API as soon as possible.

**Comment by tomchristie** [comment]
I'd also appreciate get that in soon, tho it's an API change and needs to be part of a major release.

**Comment by karpetrosyan** [comment]
Do we really need a major release to add a new public API? (when we do support the previous one)

[SEMVER](https://semver.org/) says

- **MAJOR version when you make incompatible API changes**
- MINOR version when you add functionality in a **backward compatible manner**
- PATCH version when you make backward compatible bug fixes

I thought we would release a major version if the previous API was deleted.

**Comment by karpetrosyan** [comment]
Also this:

https://semver.org/#spec-item-7
> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.

and this

https://semver.org/#spec-item-7
> Minor version Y (x.Y.z | x > 0) MUST be incremented if new, backward compatible functionality is introduced to the public API. It MUST be incremented if any public API functionality is marked as deprecated. It MAY be incremented if substantial new functionality or improvements are introduced within the private code. It MAY include patch level changes. Patch version MUST be reset to 0 when minor version is incremented.

**Comment by tomchristie** [comment]
It's an (somewhat) incompatible API change, since it introduces a deprecation warning with the current usage.

**Comment by karpetrosyan** [comment]
Deprecation cases are mentioned here, or am I missing something?

Minor version Y (x.Y.z | x > 0) MUST be incremented if new, backward compatible functionality is introduced to the public API. **It MUST be incremented if any public API functionality is marked as deprecated.** It MAY be incremented if substantial new functionality or improvements are introduced within the private code. It MAY include patch level changes. Patch version MUST be reset to 0 when minor version is incremented.

**Comment by tomchristie** [comment]
We've got a few type annotations and we're updating our dependancies. Let's roll a minor release for that.

#2879 introduces an API change and shouldn't be included. The rest is just noise.

**Comment by karpetrosyan** [comment]
Should I draft a new release? Or you want to do it yourself.

**Comment by tomchristie** [comment]
I don't mind. Do team members have the correct permissions for the ["draft a new release" control on this page](https://github.com/encode/httpx/releases)?

**Comment by karpetrosyan** [comment]
Yes, it seems I have enough permissions to do that. 
Here is the release draft: https://github.com/encode/httpx/releases/tag/untagged-3a61a71d71d75f055218

I have set the release date to tomorrow.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
-## Unreleased
+## 0.25.2 (TODO)
```
Comment (karpetrosyan) [inline CHANGELOG.md:7]:
```suggestion
## 0.25.2 (24th November, 2023)
```

**Comment by karpetrosyan** [comment]
Are you fine with releasing this @tomchristie? It seems I can't run the workflow for release.

**Comment by tomchristie** [comment]
Our process (for visibility)...

* We merge the release PR.
* We "publish release" on the release draft.
* We approve the release workflow. (requires @encode/operations permissions.)

**Comment by karpetrosyan** [comment]
Here is the 0.25.2 release: https://github.com/encode/httpx/releases/tag/0.25.2

Now it's time for @encode/operations  to double-check and run the workflow.

**Comment by tomchristie** [comment]
O hai... https://pypi.org/project/httpx/0.25.2/

**Pull request merged**